File: src/main/java/com/wisegas/grapp/domain/value/GrappStoreNodeType.java
package com.wisegas.grapp.domain.value;

public enum GrappStoreNodeType {
   REGULAR,
   ENTRANCE,
   EXIT;

   public static GrappStoreNodeType fromName(String name) {
      return Enum.valueOf(GrappStoreNodeType.class, name);
   }
}

File: src/main/webapp/app/value/GrappStoreNodeType.js
(function() {
   "use strict";

   angular.module("Grapp")
      .constant("GrappStoreNodeType", createEnum());

   function createEnum() {
      return {
         REGULAR: new GrappStoreNodeType("Regular", "content/img/marker_blue.png"),
         ENTRANCE: new GrappStoreNodeType("Entrance", "content/img/marker_green.png"),
         EXIT: new GrappStoreNodeType("Exit", "content/img/marker_red.png")
      };
   }

   function GrappStoreNodeType(displayString, iconUrl) {
      this.displayString = displayString;
      this.iconUrl = iconUrl;
   }
})();